src/Microsoft.PowerShell.Commands.Management/commands/management/ResolvePathCommand.cs

```diff
@@ -178,10 +170,23 @@ protected override void ProcessRecord()
         {
             foreach (string path in Path)
             {
+                bool popLocation = false;
                 Collection<PathInfo> result = null;
                 try
                 {
-                    result = SessionState.Path.GetResolvedPSPathFromPSPath(path, CmdletProviderContext);
+                    if (MyInvocation.BoundParameters.ContainsKey("RelativeBasePath"))
+                    {
+                        SessionState.Path.PushCurrentLocation(string.Empty);
+                        _ = SessionState.Path.SetLocation(_relativeBasePath);
+                        popLocation = true;
+                        result = SessionState.Path.GetResolvedPSPathFromPSPath(path, CmdletProviderContext);
+                        _ = SessionState.Path.PopLocation(string.Empty);
+                        popLocation = false;
```

Comment:
Why do we need `popLocation`? If it is important to restore the location here we should put `finally` here. Otherwise `finally` below will be executed in any case.

Reply:
Because objects are written to the pipeline inside the `if (_relative)` block and we don't want the temp location to affect downstream commands. I don't see any need to wrap it in another try finally block because if `GetResolvedPSPathFromPSPath` fails then we'll hit the existing finally block, and if it doesn't fail then there's obviously no need to wrap it in try/finally.

Reply:
I still believe we can remove the `popLocation`. The variable has no effect on `if (_relative)` block.

Reply:
I've moved the finally block, see latest commit.